numpy/core/src/multiarray/methods.c

```diff
@@ -1008,6 +1008,56 @@ array_getarray(PyArrayObject *self, PyObject *args)
 
 
 static PyObject *
+array_ufunc(PyArrayObject *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *ufunc, *method_name, *normal_args, *ufunc_method;
+    PyObject *result = NULL;
+
+    if (PyTuple_Size(args) < 2) {
+        PyErr_SetString(PyExc_TypeError,
+                        "__array_ufunc__ requires at least 2 arguments");
+        return NULL;
+    }
+    normal_args = PyTuple_GetSlice(args, 2, PyTuple_GET_SIZE(args));
+    if (normal_args == NULL) {
+        return NULL;
+    }
+    /* ndarray cannot handle overrides itself */
+    if (PyUFunc_WithOverride(normal_args, kwds, NULL)) {
+        result = Py_NotImplemented;
+        Py_INCREF(Py_NotImplemented);
+        goto cleanup;
+    }
+
+    ufunc = PyTuple_GET_ITEM(args, 0);
+    if (ufunc == NULL) {
```

Comment:
Right, so there's no need to check against NULL here, because if your tuple has `NULL`s in the underlying buffer, you have much bigger problems